### PR TITLE
docs: update explanation about compile from source

### PR DESCRIPTION
**Pull request by aceforeverd**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

the pr is related to #882, it add/update description for compile: 
- hardware and minimal requirements 
- troubleshooting

### CODE REVIEW

File: docs/en/compile.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
-# Build & Install
+Build & Install
+===============
+
+# Build on official compile image
```
Comment (aceforeverd) [inline docs/en/compile.md]: maybe rename this section as `Quick start`

File: docs/en/compile.md [hunk 2]
```diff
@@ -24,7 +28,38 @@
...
+- if you'd like compile thirdparty from source, checkout [third-party's requirement](../../third-party/README.md) for extra dependencies
+
+## Build OpenMLDB
+
+  Build commands are same as [Quick start](#quick-start)
+
+  ```bash
+  $ cd OpenMLDB
+  $ make
+  $ make install
```
Comment (aceforeverd) [inline docs/en/compile.md]: add `nproc` desc

File: docs/en/compile.md [hunk 3]
```diff
@@ -24,7 +28,38 @@
...
     ```
 
-### Extra Options for `make`
+# Build
+
+[build]: build
+
+## Hardware Requirements
+
+- **Memory**: 4GB RAM minimum, 8GB+ recommended.
```
Comment (aceforeverd) [inline docs/en/compile.md]: 8GB +

File: docs/en/compile.md [hunk 4]
```diff
@@ -24,7 +28,38 @@
...
+- **Disk Space**: >=25GB of free disk space for full compilation.
+- **Operating System**: A 64-bit installation of Linux or macOS >= 10.14
```
Comment (aceforeverd) [inline docs/en/compile.md]: Note about: nproc=1 current limit

**Comment by aceforeverd** [comment]
change default NPROC = 1

File: docs/cn/compile.md [hunk 5]
```diff
@@ -64,6 +97,17 @@ make CMAKE_BUILD_TYPE=Debug
...
   默认: ‘’
 
+- BUILD_BUNDLED: 从源码编译 thirdparty 依赖，而不是下载预编译包
+
+  默认: OFF
+
+## 故障排除
+
+- 如果编译机器的资源有限，例如 4G 内存, 推荐关闭默认的并行编译功能：
+- 第三方依赖 thirdparty 默认会下载在 [hybridsql-assert](https://github.com/4paradigm/hybridsql-asserts/releases) 下的预编译包，支持 CentOS 7, Ubuntu 20.04 和 macOS. 如果不在改列表中或者遇到了不在预期的链接错误，推荐从源码编译第三方库。 注意 thirdparty 编译会消耗更长的编译时间, 在2核7G内存的机器上大约需要1个小时：
```
Comment (zhanghaohit) [inline docs/cn/compile.md]: 不在`该`列表

File: docs/en/compile.md [hunk 6]
```diff
@@ -64,8 +99,25 @@ make CMAKE_BUILD_TYPE=Debug
...
+- BUILD_BUNDLED: compile thirdparty from source instead download pre-compiled
+
+  Default: OFF
+
+
+## Troubleshooting
+
+[build-troubleshooting]: build-troubleshooting
+
+- If the host machine's resource is limited, e.g a VM when 4G memory, it is advised to turn off parallel build by change the `NPROC` variable:
```
Comment (zhanghaohit) [inline docs/en/compile.md]: a VM `with` 4G memory

**Review by zhanghaohit — CHANGES_REQUESTED** [review]

File: docs/cn/compile.md [hunk 7]
```diff
@@ -64,6 +97,20 @@ make CMAKE_BUILD_TYPE=Debug
...
+
+  默认: OFF
+
+## 故障排除
+
+- 如果编译机器的资源有限，例如 4G 内存, 推荐关闭默认的并行编译功能：
+  ```bash
+  make NPROC=1
+  ```
+- 第三方依赖 thirdparty 默认会下载在 [hybridsql-assert](https://github.com/4paradigm/hybridsql-asserts/releases) 下的预编译包，支持 CentOS 7, Ubuntu 20.04 和 macOS. 如果不在改列表中或者遇到了不在预期的链接错误，推荐从源码编译第三方库。 注意 thirdparty 编译会消耗更长的编译时间, 在2核7G内存的机器上大约需要1个小时：
```
Comment (aceforeverd) [inline docs/cn/compile.md]: move troubleshooting to `Build OpenMLDB`

File: docs/cn/compile.md [hunk 8]
```diff
@@ -24,6 +28,35 @@
...
+
+- gcc 8 或更新版本
+- cmake 3.20 或更新版本
+- jdk 8
+- python3, python setuptools, python wheel
+- apache maven 3.x
+- 如果需要编译 thirdparty, 查看 [third-party's requirement](../../third-party/README.md) 里的额外要求
+
+## 编译 OpenMLDB
+
```
Comment (aceforeverd) [inline docs/cn/compile.md:63]: 简单介绍：
三种 编译 OpenMLDB 的方式

File: docs/cn/compile.md [hunk 9]
```diff
@@ -24,6 +28,35 @@
...
 
+# 编译
+
+[build]: build
+
+## 硬件要求
+
+- **内存**: 最小 4GB, 推荐 8GB+.
+- **硬盘**: 全量编译需要至少 25GB 的空闲磁盘空间
+- **操作系统**: 64 位 Linux 或者 macOS >= 10.14
```
Comment (aceforeverd) [inline docs/cn/compile.md]: troublestooting: resource limit

Comment (aceforeverd) [inline docs/cn/compile.md]: LInux -> Centos7 and Ubuntu 20.04

**Comment by aceforeverd** [comment]
docker mount rm

File: docs/cn/compile.md [hunk 10]
```diff
@@ -24,6 +28,35 @@
...
+- **硬盘**: 全量编译需要至少 25GB 的空闲磁盘空间
+- **操作系统**: 64 位 Linux 或者 macOS >= 10.14 
+
+## 依赖工具
+
+- gcc 8 或更新版本
+- cmake 3.20 或更新版本
+- jdk 8
+- python3, python setuptools, python wheel
+- apache maven 3.x
```
Comment (zhanghaohit) [inline docs/cn/compile.md]: maven has to be >= 3.3.9

see [here](https://stackoverflow.com/questions/53211941/unable-to-load-the-mojo-test-org-apache-maven-pluginsmaven-surefire-plugin)

File: docs/cn/compile.md [hunk 11]
```diff
@@ -1,29 +1,79 @@
...
+
+1. 下载 docker 镜像
     ```bash
     docker pull 4pdosc/hybridsql:0.4.0
     ```
-3. 启动docker，并绑定本地 OpenMLDB 目录到 docker 中
+
+2. 启动 docker 容器
     ```bash
     docker run -v `pwd`:/OpenMLDB -it 4pdosc/hybridsql:0.4.0 bash
```
Comment (zhanghaohit) [inline docs/cn/compile.md]: since clone is done inside docker, `-v `pwd`:/OpenMLDB` is not necessary?

**Review by zhanghaohit — CHANGES_REQUESTED** [review]

**Comment by zhanghaohit** [comment]
This PR should be merged after #875

File: docs/cn/compile.md (hunk 11 above)
Comment (aceforeverd) [reply docs/cn/compile.md]: Yes, `-v ...` should removed here, I missed this cn version.

**Comment by lumianph** [comment]
@aceforeverd i revise the Chinese version directly, please double check and modify the English version accordingly.

**Comment by aceforeverd** [comment]
> 

cool

**Review by lumianph — APPROVED** [review]

**Review by zhanghaohit — APPROVED** [review]

**Pull request merged**